classes/core/PKPContainer.php

```diff
@@ -542,6 +553,44 @@ public function isDownForMaintenance()
     {
         return Application::isUnderMaintenance();
     }
+
+    /**
+     * Determine if the application is running in the console.
+     */
+    public function runningInConsole(?string $scriptPath = null): bool
+    {
+        if (strtolower(php_sapi_name() ?: '') === 'cli') {
+            return true;
+        }
+
+        if (!$scriptPath) {
+            return false;
+        }
+
+        $scriptPath = mb_strtolower($scriptPath);
+
+        if (isset($_SERVER['SCRIPT_NAME']) && strpos(strtolower($_SERVER['SCRIPT_NAME']), $scriptPath) !== false) {
+            return true;
+        }
+
+        if (isset($_SERVER['SCRIPT_FILENAME']) && strpos(strtolower($_SERVER['SCRIPT_FILENAME']), $scriptPath) !== false) {
+            return true;
+        }
+    }
+
+    /**
+     * Get or check the current application environment.
+     */
+    public function environment(string ...$environments): string|bool
+    {
+        if (count($environments) > 0) {
+            $patterns = is_array($environments[0]) ? $environments[0] : $environments;
+
+            return Str::is($patterns, $this->get('config')['app']['env']);
```

Comment:
As you've added the type to the `...$environments`, then I think this can be turned to:

```suggestion
            return Str::is($environments, $this->get('config')['app']['env']);
```